src/plugins/expressions/public/react_expression_renderer/use_expression_renderer.ts

```diff
@@ -122,8 +131,12 @@ export function useExpressionRenderer(
 
   useEffect(() => {
     const subscription = expressionLoaderRef.current?.data$.subscribe(({ partial, result }) => {
-      setState({ isEmpty: false });
+      const newState: Partial<ExpressionRendererState> = { isEmpty: false };
+      if (!isExpressionError(result)) {
+        newState.error = null;
+      }
```

Comment:
nit: it's possible to avoid explicit typings and imperative object mutations.

```suggestion
      setState({
        isEmpty: false,
        ...(!isExpressionValueError(result) ? { error: null } : {}),
      });
```